python/pyspark/sql/pandas/conversion.py

```diff
@@ -103,10 +103,18 @@ def toPandas(self):
                     batches = self.toDF(*tmp_column_names)._collect_as_arrow()
                     if len(batches) > 0:
                         table = pyarrow.Table.from_batches(batches)
+                        del batches
                         # Pandas DataFrame created from PyArrow uses datetime64[ns] for date type
                         # values, but we should use datetime.date to match the behavior with when
                         # Arrow optimization is disabled.
-                        pdf = table.to_pandas(date_as_object=True)
+                        pandas_options = {'date_as_object': True}
+                        if self_destruct:
+                            pandas_options.update({
+                                'self_destruct': True,
+                                'split_blocks': True,
+                                'use_threads': False,
+                            })
+                        pdf = table.to_pandas(**pandas_options)
```

Comment:
Yeah, I think it's better. In which case should we disable? Maybe we should enable it by default.

Reply:
The worry is with running into things like https://github.com/pandas-dev/pandas/issues/35530 in which case the user may appreciate an escape hatch.

Reply:
yeah, I think this option can lead to other side effects, best to disable by default I think.